src/WebUI/Services/ServiceDetailsView.tsx

```diff
@@ -206,4 +224,70 @@ export class ServiceDetailsView extends BaseComponent<IServiceDetailsViewPropert
 
     private _servicesStore: ServicesStore;
     private _podsActionsCreator: PodsActionsCreator
+
+    private static _renderTypeCell(rowIndex: number, columnIndex: number, tableColumn: ITableColumn<IServiceItem>, service: IServiceItem): JSX.Element {
+        let text: string = service.type;
+        if (rowIndex == 0) {
+            text = Resources.TypeText;
+        }
+        const itemToRender = BaseKubeTable.renderColumn(text || "", BaseKubeTable.defaultColumnRenderer);
+        return BaseKubeTable.renderTableCell(rowIndex, columnIndex, tableColumn, itemToRender);
+    }
+    private static _renderClusterIPCell(rowIndex: number, columnIndex: number, tableColumn: ITableColumn<IServiceItem>, service: IServiceItem): JSX.Element {
```

Comment:
can not we club all of these in to one function and send the actual and default values?

let text: string = service.clusterIP;
        if (rowIndex == 0) {
            text = Resources.ClusterIPText;
        }
        const itemToRender = BaseKubeTable.renderColumn(text || "", BaseKubeTable.defaultColumnRenderer);
        return BaseKubeTable.renderTableCell(rowIndex, columnIndex, tableColumn, itemToRender);